python/meep.i

```diff
@@ -15,29 +15,74 @@
  *  Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
  */
 
-%module(directors="1") meep
+%module meep
 
 %{
 #define SWIG_FILE_WITH_INIT
 #include "meep/vec.hpp"
 #include "meep.hpp"
+
 using namespace meep;
+
 %}
 
 %include "numpy.i"
 
-%{
-#if NPY_API_VERSION < 0x00000007
-#define NPY_ARRAY_C_CONTIGUOUS NPY_C_CONTIGUOUS
-#define NPY_ARRAY_ALIGNED  NPY_ALIGNED
-#endif
-%}
-
 %init %{
   import_array();
 %}
 
-// TODO: apply necessary numpy typemaps
+%{
+PyObject *py_callback = NULL;
+
+static PyObject* vec2py(const meep::vec &v) {
+
+    double x = 0, y = 0, z = 0;
+
+    switch (v.dim) {
+      case meep::D1:
+        z = v.z();
+        break;
+      case meep::D2:
+        x = v.x();
+        y = v.y();
+        break;
+      case meep::D3:
+        x = v.x();
+        y = v.y();
+        z = v.z();
+        break;
+      case meep::Dcyl:
+        r = v.r();
+        z = v.z();
+        return Py_BuildValue("(ddd)", r, z, 0);
+        break;
```

Comment:
`r` is no longer declared, hence the Travis failure.  Just do `return Py_BuildValue("(ddd)", v.r(), v.z(), 0);` and skip the `r = ...` and `z = ...`.  (You can also omit `break` given the `return` clause.)